coala_quickstart/generation/Project.py

```diff
@@ -29,3 +30,66 @@ def valid_path(path: StringConverter):
     if not os.path.isdir(path):
         raise ValueError("The given path doesn't exist.")
     return path
+
+
+def language_percentage(file_paths):
+    """
+    Computes the percentage composition of each language, with unknown
+    extensions tagged with the ``Unknown`` key.
+
+    :param file_paths: A list of file paths.
+    :return:           A dict with file name as key and the percentage
+                       of occurences as the value.
+    """
+    results = {}
+
+    for file_path in file_paths:
+        ext = os.path.splitext(file_path)[1]
+        if ext in exts:
+            for lang in exts[ext]:
+                if lang in results:
+                    results[lang] += 100 / len(file_paths)
+                else:
+                    results[lang] = 100 / len(file_paths)
+
+    return results
+
+
+def get_used_languages(file_paths):
+    """
+    Identifies the most used languages in the user's project directory
+    from the files matched from the given glob expression.
+
+    :param file_paths:
+        A list of absolute file paths in the user's project directory.
+    :return:
+        A list of tuples containing a language name as the first value
+        and percentage usage in the project as the second value.
+    """
+    results = language_percentage(file_paths)
+
+    # Sort the languages based on frequency
+    return sorted(
+        results.items(),
+        key=operator.itemgetter(1),
+        reverse=True)
+
+
+def print_used_languages(printer, results):
+    """
+    Prints the sorted list of used languages along with each language's
+    percentage use.
+
+    :param printer:
+        A ``ConsolePrinter`` object used for console interactions.
+    :param results:
+        A list of tuples containing a language name as the first value
+        and percentage usage in the project as the second value.
+    """
+    if results:
```

Comment:
if not, this function shouldn't be called and the case should be handled outside, not ignored